### PR TITLE
Prevent concurrent organisation imports

### PR DESCRIPTION
**Pull request by jamiecobbett**
We should only do one of these at a time, but the cron job runs on all (3)
servers, so can trample over itself.

I suspect this is the cause of duplicate organisations in our database - the
when one instance calls find_or_create_by(slug: slug), it finds nothing, so
creates a new organisation, but another instance has done the same thing a
fraction of a second earlier.

Pattern copied from Transition: https://github.com/alphagov/transition/blob/master/lib/transition/distributed_lock.rb

Merge after: https://github.gds/gds/alphagov-deployment/pull/832

### CODE REVIEW

File: lib/contacts/distributed_lock.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+module Contacts
+  class DistributedLock
+    LIFETIME = (5 * 60) # seconds. The lock expires automatically after this time
+
+    def initialize(lock_name)
+      @lock_name = lock_name
+    end
+
+    def lock
+      redis.lock("contacts-admin:#{Rails.env}:#{@lock_name}", life: LIFETIME) do
```
Comment (alext) [inline lib/contacts/distributed_lock.rb]: If we're specifying the namespace in the config file, we should avoid hard-coding it here.

**Comment by alext** [comment]
1 minor comment, otherwise looks good to me.

File: lib/contacts/distributed_lock.rb (hunk 1 above)
Comment (jamiecobbett) [inline lib/contacts/distributed_lock.rb]: Turns out that `namespace` isn't actually a configuration option for `redis` or `mlannet-redis-lock`, so I've rebased it out of the `redis.yml` file.

Comment (alext) [inline lib/contacts/distributed_lock.rb]: Fair enough.  We could do with removing it from the version in alphagov-deployment as well, just to keep things consistent.

**Pull request merged**